daemon/container.go

```diff
@@ -81,7 +81,9 @@ func (daemon *Daemon) load(id string) (*container.Container, error) {
 	if ctr.ImagePlatform.Architecture == "" {
 		migration := daemonPlatformReader{
 			imageService: daemon.imageService,
-			content:      daemon.containerdClient.ContentStore(),
+		}
+		if daemon.containerdClient != nil {
+			migration.content = daemon.containerdClient.ContentStore()
 		}
 		migrateContainerOS(context.TODO(), migration, ctr)
```

Comment:
Other things are still needed in this case, or could this be skipped as a whole?

Reply:
No, we still want to deduce the `Architecture` from the image (but without using the containerd content store).